sklearn/metrics/pairwise.py

```diff
@@ -720,6 +720,72 @@ def haversine_distances(X, Y=None):
     return DistanceMetric.get_metric('haversine').pairwise(X, Y)
 
 
+def pairwise_wasserstein_distances(X, Y, X_d, Y_d):
+    """Computes the pairwise wasserstein distances of two groups of
+    data points with uncertainties
+    wikipedia: https://en.wikipedia.org/wiki/Wasserstein_metric
+
+    .. math::
+       D(X, Y, X_d, Y_d) = PairwiseEuclideanDistance(X, Y) +
+        sum(X_d[i]+Y_d[i]-2*sqrt(X_d[i]*Y_d[i])) for i in range(n_features)
+        for every pair of X and Y
+
+    Parameters
+    ----------
+    X    : array_like, shape (n_samples_1, n_features)
+
+    Y    : array_like, shape (n_samples_2, n_features)
+
+    X_d  : array_like, shape (n_samples_1, n_features)
+
```

Comment:
This needs description. Apart from anything else, I don't know if this is stdev, variance, or double that (error bar width).

Reply:
I will add better description.